src/Symfony/Component/Lock/Store/SemaphoreStore.php

```diff
@@ -64,21 +64,40 @@ private function lock(Key $key, bool $blocking)
         }
 
         $keyId = unpack('i', md5($key, true))[1];
-        $resource = sem_get($keyId);
```

Comment:
the ErrorException originates from a php warning
instead, I suggest using this (and removing the new method):
```php
$resource = @sem_get($keyId);
$acquired = $resource && @sem_acquire($resource, !$blocking);
```